lib/streamlit/runtime/pages_manager.py

```diff
@@ -196,19 +192,28 @@ def get_pages(self) -> dict[PageHash, PageInfo]:
     def set_pages(self, pages: dict[PageHash, PageInfo]) -> None:
         self._pages = pages
 
+    def register_pages_changed_callback(
+        self,
+        callback: Callable[[str], None],
+    ) -> Callable[[], None]:
+        # V2 strategy does not handle any pages changed event
+        return lambda: None
+
+    def invalidate_pages_cache(self) -> None:
+        pass
+
 
 class PagesManager:
     DefaultStrategy: Type[PagesStrategyV1 | PagesStrategyV2] = PagesStrategyV1
 
     def __init__(self, main_script_path, script_cache=None, **kwargs):
-        self._cached_pages: dict[PageHash, PageInfo] | None = None
-        self._pages_cache_lock = threading.RLock()
```

Comment:
I think(?) it ends up being the case that we don't need a lock here anymore, but it's not entirely obvious because it's still the case that in the V2 world we have multiple threads that may end up accessing pages simultaneously: both the user script thread and event loop thread may simultaneously access `self.pages_strategy._pages`.

I think this ends up being fine in the V2 world since only the user script thread can write to `self.pages_strategy._pages` and the event loop thread only reads, but we'll want to add some comments to clarify that the absence of locks in the V2 world is intentional + probably some additional comments in the `PagesManager` class clarifying that thread safety concerns around a strategy's cached pages are delegated to the strategy implementation

Reply:
Yea, I think pages need to be set before they are useful. We definitely reduce the risk because it's only tied to a particular session (and not shared across all sessions).